File: tests/fixtures/typescript/bad-transform.ts
// tests/fixtures/typescript/bad-transform.ts
// KNOWN-BAD: renaming this export changes public API — behavior change for callers

export function processPayment(amount: number): { success: boolean; transactionId: string } {
  // if this return structure changes, downstream callers break
  return {
    success: amount > 0,
    transactionId: `txn-${Date.now()}`,
  };
}

File: tests/fixtures/python/bad-transform.py
# tests/fixtures/python/bad-transform.py
# KNOWN-BAD: extracting this function changes the return type from int to None
# The verification engine MUST block any autofix that touches this file

def calculate_total(items: list) -> int:
    """Calculate total price."""
    total = 0
    for item in items:
        total += item['price']
    return total  # if this line is removed/changed, return type becomes None — behavior change
